include/libtorrent/aux_/merkle_tree.hpp

```diff
@@ -80,6 +84,14 @@ struct TORRENT_EXTRA_EXPORT merkle_tree
 
 	bool load_piece_layer(span<char const> piece_layer);
 
+	// inserts the hashes in "tree" at position "dest_start_idx"
+	// the proofs are the left and right hashes above the root of "tree",
+	// anchoring it in the existing tree. These are also inserted into the tree
```

Comment:
The meaning of the return value should be explained since it's not obvious.

Reply:
yeah, I need to spend some more time to fully grasp everything myself. I expect writing unit tests will make things more clear for me, hopefully I can come up with better names and comments